loader/src/main/kotlin/net/weavemc/loader/util/FileManager.kt

```diff
@@ -12,16 +12,38 @@ internal object FileManager {
 
     fun getVanillaMinecraftJar(): File {
         val os = System.getProperty("os.name").lowercase()
-        val minecraftPath = Paths.get(System.getProperty("user.home"), when {
+        val minecraftPath = when {
             os.contains("win") -> "AppData${File.separator}Roaming${File.separator}.minecraft"
             os.contains("mac") -> "Library${File.separator}Application Support${File.separator}minecraft"
             os.contains("nix") || os.contains("nux") || os.contains("aix") -> ".minecraft"
-            else -> error("Failed to retrieve Vanilla Minecraft Jar due to unsupported OS.")
-        })
+            else -> null
+        }
+        if (minecraftPath != null) {
+            val fullPath = Paths.get(System.getProperty("user.home"), minecraftPath)
+
+            val regularPath = fullPath.resolve("versions")
+                .resolve(GameInfo.version.versionName)
+                .resolve("${GameInfo.version.versionName}.jar")
+            if (regularPath.exists()) {
+                return regularPath.toFile()
+            }
+        }
 
-        return minecraftPath.resolve("versions")
-            .resolve(GameInfo.version.versionName)
-            .resolve("${GameInfo.version.versionName}.jar").toFile()
+        val gameVersion = GameInfo.version.versionName
+        val classpath = System.getProperty("java.class.path")
```

Comment:
Weave will probably always be executed on openjdk or a flavor of it, but technically this property is not necessarily set.

Reply:
I believe it is mandatory for HotSpot-based JVMs to implement this, but nonetheless I'm missing a null check thanks